test/groups/array_manifold.jl

```diff
@@ -11,15 +13,15 @@
     p, q = [exp(M, eg, hat(M, eg, ωi)) for ωi in ω]
     X = hat(M, eg, [-1.0, 2.0, 0.5])
 
-    e = Identity(AG, p)
-    @test Manifolds.array_value(e) === e
-    @test Manifolds.array_point(e) === e
+    e = make_identity(AG, p)
+    @test Manifolds.array_value(e).p == Manifolds.array_value(e.p)
+    @test_broken Manifolds.array_point(e).p == Manifolds.array_point(e.p)
```

Comment:
This might just be tricky because `array_point(e)` has to “unwrap” both `e.p` and `e` itself and create a new identity from that? Does it do that?
Is within `e` the `p` also stored as `ArrayPoint`? Otherwise the right would not make sense.

Reply:
`array_value` unwraps both `e.p` and `e` itself right now. `array_point` at the moment doesn't wrap `Identity`.

Reply:
ah, I see, then we should maybe add that double, wrap?

Reply:
Could just (roughly) be `array_point(e::Identity) = Identity(e.group, array_point(e.p))` ? Or even wrap also `e`? maybe wrapping the inner would be enough?

Reply:
I've changed `array_point(e::Identity)` to your definition. Sorry for the delay, I forgot about it.